src/IssueViz.Security.UnitTests/Taint/TaintList/TaintIssuesControlViewModelTests.cs

```diff
@@ -439,14 +466,155 @@ public void HasServerIssues_IssuesChanged_RaisesPropertyChanged()
             suppliedArgs.PropertyName.Should().Be(nameof(testSubject.HasServerIssues));
         }
 
+        [TestMethod]
+        public void SelectionChanged_SelectedIssueExistsInList_IssueSelected()
+        {
+            var issueViz = Mock.Of<IAnalysisIssueVisualization>();
+            var storeIssues = new []{ issueViz };
+            var selectionService = new Mock<IIssueSelectionService>();
+
+            var testSubject = CreateTestSubject(storeIssues, selectionService: selectionService.Object);
+
+            testSubject.SelectedIssue.Should().BeNull();
+
+            RaiseSelectionChangedEvent(selectionService, issueViz);
+
+            testSubject.SelectedIssue.Should().NotBeNull();
+            testSubject.SelectedIssue.TaintIssueViz.Should().Be(issueViz);
+        }
+
+        [TestMethod]
+        [DataRow(true)]
+        [DataRow(false)]
+        public void SelectionChanged_SelectedIssueIsNotInList_SelectionSetToNull(bool isSelectedNull)
+        {
+            var selectionService = new Mock<IIssueSelectionService>();
+
+            var testSubject = CreateTestSubject(selectionService: selectionService.Object);
+
+            testSubject.SelectedIssue.Should().BeNull();
```

Comment:
This test starts with a null selection and finishes with a null selection, so it doesn't show that the test subject is correctly setting the selection to null - it could be doing nothing at all. The test subject should start with a selection.